### DISCUSSION TITLE
SPI SCK setting

### DISCUSSION THREAD

**Discussion by Dmkrd** [earlephilhower/arduino-pico#2536] [General]
Good afternoon!
I ask for advice - 
Is it possible to set SCK high level width and SCK low level width in the SPI settings? communication with an external device via non-standard SPI is highly dependent on the pulse width. example on the screen
Thank you!
![Screenshot_13](https://github.com/user-attachments/assets/ae2775dc-f8f1-4e6c-ace7-79bd626e786e)

**Comment by earlephilhower** [comment]
Probably not, but feel free to read thru the datasheet for the SPI block registers: https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf

You might be able to write a PIO program to do it, though.  Run the PIO at several multiples of the SPI clock frequency you want, then use a sideset to create the clock while shifting in/out at appropriate.  If it's really as slow as your image (~10ms period) it shouldn't be too bad.

**Comment by Dmkrd** [comment]
Thanks a lot!!
